src/SonarScanner.MSBuild/BootstrapperClass.cs

```diff
@@ -164,8 +159,19 @@ private int PostProcess()
             }
             else
             {
-                var postProcessor = this.processorFactory.CreatePostProcessor();
-                succeeded = postProcessor.Execute(this.bootstrapSettings.ChildCmdLineArgs.ToArray(), config, teamBuildSettings);
+#if NETCOREAPP2_1
+
+                const string netcore2Warning =
+                    "From the 6th of July 2022, we will no longer release new Scanner for .NET versions that target .NET Core 2.1." +
+                    " If you are using the .NET Core Global Tool you will need to use a supported .NET runtime environment." +
+                    " For more information see https://community.sonarsource.com/t/54684";
+                WarningsSerializer.Serialize(
+                    new[] { new Warning(netcore2Warning) },
+                    Path.Combine(teamBuildSettings.SonarOutputDirectory, "AnalysisWarnings.Scanner.json"));
```

Comment:
should it be `AnalysisWarnings.S4NET.json` instead?

I am thinking that the Scanner for VSTS might leave messages in the future...

or you can leave this as is and we can think of this in the future.

Reply:
My suggestion is to merge it as is and I'll start a discussion on slack. Once we have a decision we can easily create a separate PR to update the name.
Anyway this warning will most probably not be around by the time Scanner for VSTS starts logging anything.